src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -753,14 +754,12 @@ internal static TimeSpan TicksToTimeSpan(double ticks)
 
         /// <summary>Opens a stream around the specified file descriptor and with the specified access.</summary>
         /// <param name="fd">The file descriptor.</param>
-        /// <param name="access">The access mode.</param>
+        /// <param name="direction">The pipe direction.</param>
         /// <returns>The opened stream.</returns>
-        private static FileStream OpenStream(int fd, FileAccess access)
+        private static Stream OpenStream(int fd, PipeDirection direction)
         {
             Debug.Assert(fd >= 0);
-            return new FileStream(
-                new SafeFileHandle((IntPtr)fd, ownsHandle: true),
-                access, StreamBufferSize, isAsync: false);
+            return new BufferedStream(new AnonymousPipeClientStream(direction, fd.ToString()), StreamBufferSize);
```

Comment:
Why the BufferedStream wrapper?

Reply:
It's not needed, this gets buffered already by the `StreamWriter`/`StreamReader` wrapping these `Streams`. I'll remove it.